cohortextractor/cohortextractor.py

```diff
@@ -553,20 +565,23 @@ def _make_cohort_report(input_dir, output_dir, study_name, suffix):
     logger.info(f"Created cohort report at {output_dir}/descriptives{suffix}.html")
 
 
-def dump_cohort_sql(study_definition):
-    study = load_study_definition(study_definition)
+def dump_cohort_sql(study_definition, params=()):
+    study = load_study_definition(study_definition, params=params)
     print(study.to_sql())
 
 
-def dump_study_yaml(study_definition):
-    study = load_study_definition(study_definition)
+def dump_study_yaml(study_definition, params=()):
+    study = load_study_definition(study_definition, params=params)
     print(yaml.dump(study.to_data()))
 
 
-def load_study_definition(name, value="study"):
+def load_study_definition(name, value="study", params=()):
     sys.path.extend([relative_dir(), os.path.join(relative_dir(), "analysis")])
     # Avoid creating __pycache__ files in the analysis directory
     sys.dont_write_bytecode = True
+    # Set any supplied "<key>=<value>" parameters on the global `params` dict
+    cohortextractor.params.clear()
+    cohortextractor.params.update(param.partition("=")[::2] for param in params)
```

Comment:
Would it be useful to log what the parameters are at this point?  If a user's done something weird, like `--param key,val` instead of `--param key=val`, the param will be interpreted as `{"key,val": ""}` and possibly not immediately obvious what they did wrong?

Reply:
:+1: I've added logging now